--- a/security/security.c
+++ b/security/security.c
@@ -856,6 +856,7 @@ int security_bprm_creds_for_exec(struct linux_binprm *bprm)
 {
 	return call_int_hook(bprm_creds_for_exec, 0, bprm);
 }
+EXPORT_SYMBOL(security_bprm_creds_for_exec);
 
 int security_bprm_creds_from_file(struct linux_binprm *bprm, struct file *file)
 {
@@ -871,6 +872,7 @@ int security_bprm_check(struct linux_binprm *bprm)
 		return ret;
 	return ima_bprm_check(bprm);
 }
+EXPORT_SYMBOL(security_bprm_check);
 
 void security_bprm_committing_creds(struct linux_binprm *bprm)
 {
@@ -1225,6 +1227,7 @@ int security_path_truncate(const struct path *path)
 		return 0;
 	return call_int_hook(path_truncate, 0, path);
 }
+EXPORT_SYMBOL(security_path_truncate);
 
 int security_path_chmod(const struct path *path, umode_t mode)
 {
@@ -1252,6 +1255,7 @@ int security_inode_create(struct inode *dir, struct dentry *dentry, umode_t mode
 		return 0;
 	return call_int_hook(inode_create, 0, dir, dentry, mode);
 }
+EXPORT_SYMBOL(security_inode_create);
 EXPORT_SYMBOL_GPL(security_inode_create);
 
 int security_inode_link(struct dentry *old_dentry, struct inode *dir,
@@ -1261,6 +1265,7 @@ int security_inode_link(struct dentry *old_dentry, struct inode *dir,
 		return 0;
 	return call_int_hook(inode_link, 0, old_dentry, dir, new_dentry);
 }
+EXPORT_SYMBOL(security_inode_link);
 
 int security_inode_unlink(struct inode *dir, struct dentry *dentry)
 {
@@ -1268,6 +1273,7 @@ int security_inode_unlink(struct inode *dir, struct dentry *dentry)
 		return 0;
 	return call_int_hook(inode_unlink, 0, dir, dentry);
 }
+EXPORT_SYMBOL(security_inode_unlink);
 
 int security_inode_symlink(struct inode *dir, struct dentry *dentry,
 			    const char *old_name)
@@ -1276,6 +1282,7 @@ int security_inode_symlink(struct inode *dir, struct dentry *dentry,
 		return 0;
 	return call_int_hook(inode_symlink, 0, dir, dentry, old_name);
 }
+EXPORT_SYMBOL(security_inode_symlink);
 
 int security_inode_mkdir(struct inode *dir, struct dentry *dentry, umode_t mode)
 {
@@ -1283,6 +1290,7 @@ int security_inode_mkdir(struct inode *dir, struct dentry *dentry, umode_t mode)
 		return 0;
 	return call_int_hook(inode_mkdir, 0, dir, dentry, mode);
 }
+EXPORT_SYMBOL(security_inode_mkdir);
 EXPORT_SYMBOL_GPL(security_inode_mkdir);
 
 int security_inode_rmdir(struct inode *dir, struct dentry *dentry)
@@ -1291,6 +1299,7 @@ int security_inode_rmdir(struct inode *dir, struct dentry *dentry)
 		return 0;
 	return call_int_hook(inode_rmdir, 0, dir, dentry);
 }
+EXPORT_SYMBOL(security_inode_rmdir);
 
 int security_inode_mknod(struct inode *dir, struct dentry *dentry, umode_t mode, dev_t dev)
 {
@@ -1298,6 +1307,7 @@ int security_inode_mknod(struct inode *dir, struct dentry *dentry, umode_t mode,
 		return 0;
 	return call_int_hook(inode_mknod, 0, dir, dentry, mode, dev);
 }
+EXPORT_SYMBOL(security_inode_mknod);
 
 int security_inode_rename(struct inode *old_dir, struct dentry *old_dentry,
 			   struct inode *new_dir, struct dentry *new_dentry,
@@ -1317,6 +1327,7 @@ int security_inode_rename(struct inode *old_dir, struct dentry *old_dentry,
 	return call_int_hook(inode_rename, 0, old_dir, old_dentry,
 					   new_dir, new_dentry);
 }
+EXPORT_SYMBOL(security_inode_rename);
 
 int security_inode_readlink(struct dentry *dentry)
 {
@@ -1352,6 +1363,7 @@ int security_inode_setattr(struct user_namespace *mnt_userns,
 		return ret;
 	return evm_inode_setattr(mnt_userns, dentry, attr);
 }
+EXPORT_SYMBOL(security_inode_setattr);
 EXPORT_SYMBOL_GPL(security_inode_setattr);
 
 int security_inode_getattr(const struct path *path)
@@ -1360,6 +1372,7 @@ int security_inode_getattr(const struct path *path)
 		return 0;
 	return call_int_hook(inode_getattr, 0, path);
 }
+EXPORT_SYMBOL(security_inode_getattr);
 
 int security_inode_setxattr(struct user_namespace *mnt_userns,
 			    struct dentry *dentry, const char *name,
@@ -1537,6 +1550,7 @@ int security_file_permission(struct file *file, int mask)
 
 	return fsnotify_perm(file, mask);
 }
+EXPORT_SYMBOL(security_file_permission);
 
 int security_file_alloc(struct file *file)
 {
@@ -1684,6 +1698,7 @@ int security_file_open(struct file *file)
 
 	return fsnotify_perm(file, MAY_OPEN);
 }
+EXPORT_SYMBOL(security_file_open);
 
 int security_task_alloc(struct task_struct *task, unsigned long clone_flags)
 {
@@ -1832,6 +1847,7 @@ int security_task_fix_setuid(struct cred *new, const struct cred *old,
 {
 	return call_int_hook(task_fix_setuid, 0, new, old, flags);
 }
+EXPORT_SYMBOL(security_task_fix_setuid);
 
 int security_task_fix_setgid(struct cred *new, const struct cred *old,
 				 int flags)
@@ -1938,6 +1954,7 @@ int security_task_prctl(int option, unsigned long arg2, unsigned long arg3,
 	}
 	return rc;
 }
+EXPORT_SYMBOL(security_task_prctl);
 
 void security_task_to_inode(struct task_struct *p, struct inode *inode)
 {
